activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -509,7 +509,7 @@ def clear_schema_cache(filename)
       def with_temporary_pool_for_each(env: ActiveRecord::Tasks::DatabaseTasks.env, name: nil, clobber: false, &block) # :nodoc:
         if name
           db_config = ActiveRecord::Base.configurations.configs_for(env_name: env, name: name)
-          with_temporary_pool(db_config, clobber: clobber, &block)
+          with_temporary_pool(db_config, clobber: clobber, &block) if db_config
```

Comment:
We can't make this change, because it means that passing a typo-d or incorrect config name would silently fail for any rake task relying on `with_temporary_pool_for_each`. This method has to return an error if the db doesn't exist. Skipping needs to be handled in the task, not this method.

Reply:
Makes sense, thanks 👍 Added a check to the rake task itself.